src/atoms/Text/Text.native.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+import theme from '../../tokens/theme';
+import { Text as NativeText } from 'react-native';
+import { getAllColorNames, resolveColorFromString } from '../../utils/colors';
+import automation from '../../_helpers/automation-attributes';
+
+const fontFamilies = {
+  light: `Lato-Regular`,
+  regular: `Lato-Regular`,
+  bold: `Lato-Bold`,
+};
+
+const StyledText = styled(NativeText)(
+  (props) => `
+  font-family: ${fontFamilies[props.weight]};
+  font-size: ${props.theme.fonts.size[props.size]};
+  color: ${resolveColorFromString(props.theme, props.color) || props.theme.colors.shade[800]};
+  text-decoration-line: ${props.underline ? 'underline' : 'none'};
+  align-self: ${props.align};
+`,
+);
+
+const Text = ({ weight, size, underline, align, testID, color, children }) => {
+  return (
```

Comment:
shouldn't we expose just `size` and `color`? Is it because button has some weird styles that's why we are exposing everything?

Reply:
`<Button />` will most probably need `size`, `color` and `weight`.

Reply:
Yes

Reply:
Now the more I think about it the more I get inclined towards private and public props. The pattern we can follow is
`_privatepropName`
`publicPropName`
anything that starts with underscore is private and if someone uses it(consumer apps) then it will be easier to catch in PR reviews. With that we can make our consumer API much more leaner and also have flexobility.

<Text
   size="small"
   color="azzure.100"
   _weight="bold"
  _isUnderlined={true}
/>

So the consumers can get the max benefit without writing size weight and what not everytime.